src/hdf5plugin/_filters.py

```diff
@@ -466,34 +465,39 @@ class SZ(_FilterRefClass):
     filter_name = "sz"
     filter_id = SZ_ID
 
-    def __init__(self, absolute=None, relative=None, pointwise_relative=1e-05):
+    def __init__(self, absolute=None, relative=None, pointwise_relative=None):
+        if (absolute, relative, pointwise_relative).count(None) < 2:
+            raise TypeError("hdf5plugin.SZ() takes at most one not None argument")
+
         # Get SZ encoding options
         if absolute is not None:
             sz_mode = 0
-            parameter = absolute
         elif relative is not None:
             sz_mode = 1
-            parameter = relative
-        elif pointwise_relative is not None:
-            sz_mode = 10
-            parameter = pointwise_relative
         else:
-            raise TypeError("One of the options need to be provided: absolute, relative or pointwise_relative.")
+            sz_mode = 10
+            if pointwise_relative is None:
+                pointwise_relative = 1e-5
 
-        packed_error = self.pack_error(parameter)
-        compression_opts = (sz_mode, *packed_error, *packed_error, *packed_error, *packed_error)
+        compression_opts = (
+            sz_mode,
+            *self.__pack_float64(absolute or 0.),
+            *self.__pack_float64(relative or 0.),
+            *self.__pack_float64(pointwise_relative or 0.),
+            *self.__pack_float64(0.),  # psnr
+        )
```

Comment:
Previous implementation was working because not all options are used depending on the mode, but this better matches the meaning of the `compression_opts`